### DISCUSSION TITLE
Is there a way that the drawer doesn't affect math.random?

### DISCUSSION THREAD

**Discussion by Paraxdev** [mantinedev/mantine#121] [Q&A] [answered]
I have a simple TicTacToe game, and it uses math.random to choose the players... however.
If I spam the drawer open it will change the text but since I use p5 to draw stuff it doesn't change it there, so I would love to know if this is a bug or something that can be disabled?

ohh and also what is it with Divider and Hr?
I use Hr but seems like the docs just has Divider and not Hr
is it intentional?

**Comment by Paraxdev** [comment]
Also, I have the error where the notification moves the canvas from p5 up:
![why](https://disq.me/HqudRk.gif)

**Reply by rtivital** [reply]
Most likely it's not the issue with notifications, try making your canvas fixed width and height to avoid this

**Comment by rtivital** [comment, marked as answer]
> If I spam the drawer open it will change the text but since I use p5 to draw stuff it doesn't change it there, so I would love to know if this is a bug or something that can be disabled?

Can you show what happens here? Cannot understand based on description

> I use Hr but seems like the docs just has Divider and not Hr

Hr component is deprecated and replaced with Divider, Hr will be removed

**Reply by Paraxdev** [reply]
![O and X Switch on drawer open](https://disq.me/GZaJGb.gif)
the X and O gets regenerated every time the drawer opens

**Reply by Paraxdev** [reply]
> Hr component is deprecated and replaced with Divider, Hr will be removed 

Isn't it the same?

**Reply by rtivital** [reply]
> the X and O gets regenerated every time the drawer opens

My guess is that you generate these values in component on each render, to fix that save value to state

**Reply by rtivital** [reply]
>Isn't it the same?

Hr stands for Horizontal rule, Divider includes optional label and can be used in vertical orientation
